lib/execution_engine2/utils/job_requirements_resolver.py

```diff
@@ -344,3 +354,153 @@ def _has_value(cls, inc):
+        dm = args[8] if (args[8] is not None) else reqs.pop(DEBUG_MODE, None)
+
+        return JobRequirements(
+            args[0] or reqs[REQUEST_CPUS],
+            args[1] or reqs[REQUEST_MEMORY],
+            args[2] or reqs[REQUEST_DISK],
+            client_group,
+            client_group_regex=cgr,
+            bill_to_user=args[5],
+            ignore_concurrency_limits=args[6],
+            scheduler_requirements=scheduler_requirements,
+            debug_mode=dm,
+        )
+
+    def _get_client_group(self, user_cg, catalog_cg, module_name, function_name):
+        cg = next(
+            i
+            for i in [
+                self._override_client_group,
+                user_cg,
+                catalog_cg,
+                self._default_client_group,
+            ]
+            if i is not None
+        )
+        if cg not in self._clientgroup_default_configs:
+            if cg == catalog_cg:
+                raise IncorrectParamsException(
+                    f"Catalog specified illegal client group '{cg}' for method "
+                    + f"{module_name}.{function_name}"
+                )
+            raise IncorrectParamsException(f"No such clientgroup: {cg}")
+        return cg
+
+    def _get_catalog_reqs(self, module_name, function_name):
+        # could cache results for 30s or so to speed things up... YAGNI
+        group_config = self._catalog.list_client_group_configs(
+            {"module_name": module_name, "function_name": function_name}
+        )
+        # If group_config is empty, that means there's no clientgroup entry in the catalog
+        # It'll return an empty list even for non-existent modules
+        if not group_config:
+            return {}
+        if len(group_config) > 1:
+            raise ValueError(
+                "Unexpected result from the Catalog service: more than one client group "
+                + f"configuration found for method {module_name}.{function_name}"
+            )
+
+        resources_request = group_config[0].get(_CLIENT_GROUPS, None)
+
+        # No client group provided
+        if not resources_request:
+            return {}
+        # JSON
+        if "{" in resources_request[0]:
+            try:
+                rv = json.loads(", ".join(resources_request))
+            except ValueError:
+                raise ValueError(
+                    "Unable to parse JSON client group entry from catalog "
+                    + f"for method {module_name}.{function_name}"
+                )
+            return {k.strip(): rv[k] for k in rv}
+        # CSV Format
+        rv = {CLIENT_GROUP: resources_request.pop(0)}
+        for item in resources_request:
+            if "=" not in item:
+                raise ValueError(
+                    f"Malformed requirement. Format is <key>=<value>. Item is '{item}' for "
+                    + f"catalog method {module_name}.{function_name}"
+                )
+            (key, value) = item.split("=")
```

Comment:
should this be limited so it only splits once? (i.e. it splits on the first `=` to produce a key and a value)

Reply:
Good call, fixed